includes/api/class-wc-rest-product-variations-controller.php

```diff
@@ -248,11 +249,17 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 
 		$variation->set_parent_id( absint( $request['product_id'] ) );
 
-		// Status.
+		// TODO Fix/Remove in future API version bump. This controls the wrong thing.
+		// https://github.com/woocommerce/woocommerce/issues/15215
 		if ( isset( $request['visible'] ) ) {
 			$variation->set_status( false === $request['visible'] ? 'private' : 'publish' );
 		}
 
+		// Status.
+		if ( isset( $request['status'] ) ) {
+			$variation->set_status( get_post_status_object( $request['status'] ) ? $request['status'] : 'draft' );
```

Comment:
I think it would be useful. I have one use-case in mind where this would be valuable (even though core doesn't use it yet). In the interface I am working on, if attributes/variation types change, we invalidate existing variations that don't match the new format. I am thinking I would like to mark these as draft instead of immediately trashing them, so they can easily be recovered.

Separate from that, this would 100% match the root products endpoint too.